app/controllers/api/v2/rgb_top_holders_controller.rb

```diff
@@ -0,0 +1,45 @@
+module Api
+  module V2
+    class RgbTopHoldersController < BaseController
+      def show
+        expires_in 15.minutes, public: true, stale_while_revalidate: 5.minutes, stale_if_error: 5.minutes
+
+        udt = Udt.find_by(udt_type: %i[xudt xudt_compatible], type_hash: params[:id])
+        return head :not_found unless udt
+
+        merged_array = btc_top_holders(udt) + ckb_top_holders(udt)
+        top10 = merged_array.sort_by { |item| -item[:amount].to_f }.take(10)
+
+        render json: { data: top10 }
+      end
+
+      private
+
+      def btc_top_holders(udt)
+        result = BitcoinAddressMapping.
+          joins("LEFT OUTER JOIN udt_accounts ON udt_accounts.address_id = bitcoin_address_mappings.ckb_address_id").
+          where(udt_accounts: { udt_id: udt.id }).
```

Comment:
Are accounts with zero balance in this table? If so, append a filter amount > 0  could speed up this query

Reply:
> Are accounts with zero balance in this table? If so, append a filter amount > 0 could speed up this query

Added filter condition: `where("udt_accounts.amount > 0")`